test/unit/specs/components/governance/PageProposal.spec.js

```diff
@@ -10,43 +15,153 @@ let status = {
 }
 
 describe(`PageProposal`, () => {
-  let wrapper
-  let { mount } = setup()
-
-  beforeEach(() => {
-    let instance = mount(PageProposal, {
-      propsData: {
-        proposal,
-        status
-      }
+  describe(`with magic`, () => {
+    let wrapper
+    let { mount, localVue } = setup()
+    localVue.use(Vuelidate)
+
+    const $store = {
+      commit: jest.fn(),
+      dispatch: jest.fn()
+    }
+
+    beforeEach(() => {
+      let instance = mount(PageProposal, {
+        localVue,
+        propsData: {
+          proposal,
+          status
+        },
+        $store
+      })
+      wrapper = instance.wrapper
+      wrapper.update()
     })
-    wrapper = instance.wrapper
-    wrapper.update()
-  })
 
-  it(`has the expected html structure`, async () => {
-    // after importing the @tendermint/ui components from modules
-    // the perfect scroll plugin needs a $nextTick and a wrapper.update
-    // to work properly in the tests (snapshots weren't matching)
-    // this has occured across multiple tests
-    await wrapper.vm.$nextTick()
-    wrapper.update()
-    expect(wrapper.vm.$el).toMatchSnapshot()
-  })
+    it(`has the expected html structure`, async () => {
+      // after importing the @tendermint/ui components from modules
+      // the perfect scroll plugin needs a $nextTick and a wrapper.update
+      // to work properly in the tests (snapshots weren't matching)
+      // this has occured across multiple tests
+      await wrapper.vm.$nextTick()
+      wrapper.update()
+      expect(wrapper.vm.$el).toMatchSnapshot()
+    })
+
+    it(`should return the block number`, () => {
+      expect(wrapper.vm.voteBlock).toBe(`block #135`)
+    })
+
+    it(`should return the end of the sentence`, () => {
+      proposal.submit_block = `135`
+      let { wrapper } = mount(PageProposal, {
+        propsData: {
+          proposal,
+          status
+        }
+      })
+      wrapper.update()
+      expect(wrapper.vm.voteBlock).toBe(`the same block`)
+    })
+
+    describe(`Modal onVote`, () => {
+      it(`enables voting if the proposal is Active`, () => {
+        let status = { button: `vote` }
```

Comment:
just thinking: status seems to be a complex object that doesn't seem self explanatory. Maybe we could split the object in a refactor.

Reply:
cc: @jbibla ^^